include/dfsinternal.h

```diff
@@ -24,10 +24,12 @@
 /** @brief Representation of a directory entry */
 struct directory_entry
 {
-    /** @brief File size and flags.  See #FLAGS_FILE, #FLAGS_DIR and #FLAGS_EOF */
-    uint32_t flags;
+    /** @brief File size */
+    uint32_t size;
```

Comment:
That's true, but each directory_entry is save in each own sector, so it's still padded out to 512 bytes no matter what we decide to change here.

But I'm OK with packing flags in upper bits anyhow. If we do, we can restore MAX_FILENAME_LEN to the previous value, and maybe at this point we can also decrease the sector size to 256 bytes rather than 512, so that we actually benefit from the packing we're performing.

Does that sound OK?

Reply:
Oh sorry, forget about that, the sector *is* 256 bytes already! I'm not sure why I was thinking 512.

OK I can pack the flags back to 512 and maybe add an assert that directory_entry always fits a sector.

Reply:
Assert is a good idea! Thanks for the change!